ui/pages/SummaryData/reducers.js

```diff
@@ -43,8 +43,9 @@ export const loadSuccessStory = successStoryTypes => (dispatch) => {
 
 export const loadSavedVariants = ({ tag, gene = '' }) => (dispatch, getState) => {
   // Do not load if already loaded
+  const stateKey = tag && `${tag.split(TAG_URL_DELIMITER).sort().join(TAG_URL_DELIMITER)}${gene}`
   if (tag) {
-    if (getState().savedVariantTags[tag]) {
+    if (getState().savedVariantTags[stateKey]) {
```

Comment:
Add the gene to the key for the state so that no states have to be removed when gene changed.

Reply:
this seems to be adding a lof of complexity  that will be hard to maintain and have almost no performance improvement - we don't anticipate users toggling back and forth to the same gene, so refetching a gene is not an issue. Please explain why this was not needed before but is now needed because we are adding support for multiple tags

Reply:
It was a bug before when the gene changed. The UI still displayed the variant data for the previous gene.

Reply:
then aither a) make a separate PR to fix the bug and branch from that. or b) make a ticket to fix that bug and fix it later and not as part of this PR. This PR is really hard to understand because there is a lot in here that is unrelated to the functionality change requested in the ticket, and I can't review it effectively

Reply:
I can leave the reducer unchanged for this PR and create a bug ticket.